research_town/envs/env_proposal_writing_without_rag.py

```diff
@@ -29,8 +29,14 @@ def __init__(
     @beartype
     def on_enter(self, **context: Any) -> None:
         # Assign leader and members from context or sample them
-        self.leader = context.get('leader', self.agent_manager.sample_leader())
-        self.members = context.get('members', self.agent_manager.sample_members())
+        if context.get('leader') is None:
```

Comment:
can you help change all the things in all environment files here? this is a potential bug in research-town